packages/devextreme/testing/tests/DevExpress.ui.widgets.editors/fileUploader.tests.js

```diff
@@ -4338,3 +4338,36 @@ QUnit.module('readOnly option', moduleConfig, () => {
 
 });
 
+QUnit.module('dxButton integration', moduleConfig, () => {
+    QUnit.test('dialog should be shown after press enter key on dxButton (T1178836)', function(assert) {
+        if(devices.real().deviceType !== 'desktop') {
+            assert.ok(true, 'keyboard is not supported for not generic devices');
+            return;
+        }
+
+        const $customDialogTrigger = $('<div>').addClass('trigger').appendTo('#qunit-fixture');
+
+        $('.trigger').dxButton({
+            text: 'button',
+            useSubmitBehavior: true
+        });
+
+        const instance = $('#fileuploader').dxFileUploader({
+            dialogTrigger: $customDialogTrigger,
+            visible: false
+        }).dxFileUploader('instance');
+
+        sinon.stub(instance, '_selectButtonClickHandler');
```

Comment:
Cannot we get rid of the using of a private api?
For example, check that input's click handler is called

Reply:
No. Here we want to test the interaction between the button and the uploader. And the uploader does not have suitable public methods.